examples/non_cartesian_reconstruction.py

```diff
@@ -0,0 +1,97 @@
+"""
+Neuroimaging non-cartesian reconstruction
+=========================================
+
+Author: Chaithya G R
+
+In this tutorial we will reconstruct an MRI image from non-cartesian kspace
+measurements.
+
+Import neuroimaging data
+------------------------
+
+We use the toy datasets available in pysap, more specifically a 2D brain slice
+and the acquisition cartesian scheme.
+"""
+
+# Package import
+from mri.numerics.fourier import NFFT
+from mri.numerics.reconstruct import sparse_rec_fista
+from mri.numerics.utils import generate_operators
+from mri.numerics.utils import convert_locations_to_mask
+from mri.parallel_mri.extract_sensitivity_maps import \
+    gridded_inverse_fourier_transform_nd
+import pysap
+from pysap.data import get_sample_data
+
+# Third party import
+from modopt.math.metrics import ssim
+import numpy as np
+
+# Loading input data
+image = pysap.utils.load_image('../../../Data/Pysap_examples/base_image.npy')
+
+# Obtain MRI non-cartesian mask
+radial_mask = get_sample_data("mri-radial-samples")
+kspace_loc = radial_mask.data
+mask = pysap.Image(data=convert_locations_to_mask(kspace_loc, image.shape))
+
+# View Input
+image.show()
+mask.show()
+
+#############################################################################
+# Generate the kspace
+# -------------------
+#
+# From the 2D brain slice and the acquisition mask, we retrospectively
+# undersample the k-space using a radial acquisition mask
+# We then reconstruct the zero order solution as a baseline
+
+# Get the locations of the kspace samples and the associated observations
+fourier_op = NFFT(samples=kspace_loc, shape=image.shape)
+kspace_obs = fourier_op.op(image.data)
+
+# Gridded solution
+grid_space = np.linspace(-0.5, 0.5, num=image.shape[0])
+grid2D = np.meshgrid(grid_space, grid_space)
+grid_soln = gridded_inverse_fourier_transform_nd(kspace_loc, kspace_obs,
+                                                 tuple(grid2D), 'linear')
+image_rec0 = pysap.Image(data=grid_soln)
+image_rec0.show()
+base_ssim = ssim(image_rec0, image)
+print('The Base SSIM is : ' + str(base_ssim))
+
+#############################################################################
+# FISTA optimization
+# ------------------
+#
+# We now want to refine the zero order solution using a FISTA optimization.
+# The cost function is set to Proximity Cost + Gradient Cost
+
+# Generate operators
+gradient_op, linear_op, prox_op, cost_op = generate_operators(
+    data=kspace_obs,
+    wavelet_name=24,
+    samples=kspace_loc,
+    mu=6.45e-07,
```

Comment:
would `6 * 1e-7` work ?

Reply:
Will check and report change in SSIM here. If it is not significant, Ill make the change

Reply:
SSIM is now, 0.9210 it was approx same, changed.